mavros_extras/src/plugins/odom.cpp

```diff
@@ -142,52 +81,25 @@ class OdometryPlugin : public plugin::PluginBase {
 	ros::Publisher odom_pub;			//!< nav_msgs/Odometry publisher
 	ros::Subscriber odom_sub;			//!< nav_msgs/Odometry subscriber
 
-	std::string local_frame_in;			//!< orientation and source of the input local frame
-	std::string local_frame_out;			//!< orientation and source of the output local frame
-	std::string local_frame_orientation_in;		//!< orientation of the local frame (input data)
-	std::string local_frame_orientation_out;	//!< orientation of the local frame (output data)
-	std::string body_frame_orientation_in_desired;	//!< orientation of the body frame (input data)
-	std::string body_frame_orientation_out_desired;	//!< orientation of the body frame (output data)
-	std::string frame_id;				//!< parent frame identifier
-	std::string child_frame_id;			//!< child frame identifier
-
-	MAV_FRAME lf_id;				//!< local frame (pose) ID
-	MAV_FRAME bf_id;				//!< body frame (pose) ID
+	std::string fcu_odom_parent_id_des;			//!< desired orientation of the child frame (input data)
+	std::string fcu_odom_child_id_des;			//!< desired orientation of the body frame (input data)
+	std::string ext_odom_parent_id;				//!< parent frame of the ext odometry msg
+	std::string ext_odom_child_id;				//!< child frame of the ext odometry msg
```

Comment:
This information is already in the Odometry message. Why do we need to set the frame manually again?

Reply:
But the frame names differs between PX4 and ROS side. e.g. we receive odom message from fcu. parent frame will be local_origin(_ned) and we want to look up transform from map_ned to map.

Reply:
Yes, but I think we don't need it for the ext odom one, because the frame information already comes in the message, and that is what we should send to FCU.

Reply:
Yes, you are right. This is ros side only